### PR TITLE
Fixes #1943 - Call reviveOffers when already declined offers

### LINKED ISSUE #1943: Make the revive offer patch available on master branch

### PR DESCRIPTION
**Pull request by kolloch**
are potentially of use.

Technical debt:

* Decompose AppTaskLauncherActor since it gets to many responsibilities

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala [hunk 1]
```diff
@@ -82,10 +91,4 @@ class CoreModuleImpl @Inject() (
 
   // FLOW CONTROL GLUE
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala]: Shouldn't we remove this comment? As well as the previous one.

File: src/main/scala/mesosphere/marathon/core/flow/FlowModule.scala [hunk 2]
```diff
@@ -24,12 +27,23 @@ class FlowModule(leadershipModule: LeadershipModule) {
     */
   def reviveOffersWhenOfferMatcherManagerSignalsInterest(
     clock: Clock, conf: ReviveOffersConfig,
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/core/flow/FlowModule.scala]: nit: should these go in different lines?

File: src/main/scala/mesosphere/marathon/core/flow/FlowModule.scala [hunk 3]
```diff
@@ -24,12 +27,23 @@ class FlowModule(leadershipModule: LeadershipModule) {
...
-    offersWanted: Observable[Boolean], driverHolder: MarathonSchedulerDriverHolder): Unit = {
-    lazy val reviveOffersActor = ReviveOffersActor.props(
-      clock, conf,
-      offersWanted, driverHolder
-    )
-    leadershipModule.startWhenLeader(reviveOffersActor, "reviveOffersWhenWanted")
+    marathonEventStream: EventStream,
+    offersWanted: Observable[Boolean],
+    driverHolder: MarathonSchedulerDriverHolder): Option[OfferReviver] = {
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/core/flow/FlowModule.scala]: shouldn't we change the method name to reflect that this now returns an Option[OfferReviver]?

**Comment by gkleiman** [comment]
LGTM, only a few very minor comments. Will happily merge once they're addressed.

**Comment by gkleiman** [comment]
I think we should also mention the new flag (`--disable_revive_offers_for_new_apps`) in changelog.md

**Comment by kolloch** [comment]
I added the changelog entry and rebased this pull request.

**Comment by gkleiman** [comment]
Merged!

**Pull request merged**